packages/core/lib/logger.js

```diff
@@ -1,96 +1,56 @@
-var winston = require('winston');
 var format = require('date-fns/format');
 
-var logger;
-var xrayLogLevel = process.env.AWS_XRAY_LOG_LEVEL;
+var validLogLevels = [ 'trace', 'debug', 'info', 'warn', 'error'  ];
+var logLevel = calculateLogLevel(process.env.AWS_XRAY_DEBUG_MODE ? 'debug' : process.env.AWS_XRAY_LOG_LEVEL);
 
-if (process.env.AWS_XRAY_DEBUG_MODE) {
-  logger = new (winston.Logger)({
-    transports: [
-      new (winston.transports.Console)({
-        formatter: outputFormatter,
-        level: 'debug',
-        timestamp: timestampFormatter
-      })
-    ]
-  });
-} else if (xrayLogLevel) {
-  logger = new (winston.Logger)({
-    transports: [
-      new (winston.transports.Console)({
-        formatter: outputFormatter,
-        level: xrayLogLevel,
-        timestamp: timestampFormatter
-      })
-    ]
-  });
-} else {
-  logger = new (winston.Logger)({});
-}
+var logger = {
+  error: createLoggerForLevel('error'),
+  info: createLoggerForLevel('info'),
+  warn: createLoggerForLevel('warn'),
+  debug: createLoggerForLevel('debug'),
+  setLevel: (level) => logLevel = calculateLogLevel(level),
```

Comment:
To take a step back, immediately after writing that set level method I kind of regretted it because I’m not entirely sure what the use case for it is. Particularly as the other methods are fairly comply implemented (take the winston logger for example) but setLevel isn’t.

So I suppose I’m asking do you think it is worth keeping?

Reply:
I agree that I can't really imagine a use case of needing to set the log level programmatically instead of with an EV. I have no issues with scrapping it.

Reply:
Thanks for the feedback. Given how late it is here I’ll update the code as requested and push up a new version on or before Monday morning. Have a nice weekend.

Reply:
Have removed the setLevel method as discussed so the environment variables are the only way to set the level of the logger.